File: Matlab/scripts/Download_Data_Yahoo.m
%% Download data from Yahoo
addpath(genpath(cd))
%% Anders Dyhr Sandgaard
clc
% Example Call: Fetching AMD stock data
symbol = 'AAPL';
start_dt = '1-Feb-2010';
end_dt = '7-Feb-2026'; % Today
interval = '1m'; %
try
    amdData = getMarketDataViaYahoo(symbol, start_dt, end_dt, interval);
    
    % Display the first few rows
    disp('Successfully retrieved data:')
    head(amdData)
    
    % Quick visualization
    figure;
    plot(amdData.Date, amdData.Close, 'LineWidth', 1.5);
    grid on;
    title(['Stock Price History: ', symbol]);
    xlabel('Date');
    ylabel('Closing Price (USD)');
    
catch ME
    fprintf('Error: %s\n', ME.message);
end

File: Python/scripts/Download_data_Yahoo.py
# %%
from GetMarketDataYahoo import GetMarketDataYahoo
from PerformanceMetrics import CalcSharpeRatio
import matplotlib.pyplot as plt
# %%
symbols = ['IGE']
start_dt = '2001-11-26' 
end_dt = '2007-11-15'
interval = '1h'  # Daily data
data = GetMarketDataYahoo(symbols, start_dt, end_dt, interval);

# %% 3. Plotting
data[symbols[0]]['Close'].plot(
    figsize=(20, 10),
    title=f'{symbols[0]} Hourly Close Price', 
    grid=True, color='blue', linewidth=1.5)
plt.xlabel('Date/Time')     

# %% calculate sharpe ratio
sharpe = CalcSharpeRatio(data, price_col='Close');
print(sharpe)